src/flint/test/test_all.py

```diff
@@ -3075,19 +3142,29 @@ def quick_poly():
         # # XXX: Not sure what this should do in general:
         assert raises(lambda: pow(P(1, ctx=ctx), 2, 3), NotImplementedError)
 
-        if is_field:
-            assert (f * g).gcd(f) == f / 4
+        if (P is not flint.fmpz_mod_mpoly and P is not flint.nmod_mpoly) or f.context().is_prime():
```

Comment:
Give the number of places where this condition is checked it should probably have a name like:
```
composite_characteristic = (P in (fmpz_mod_mpoly, nmod_mpoly)) and not P.is_prime()
```